core/src/main/scala/kafka/coordinator/group/CoordinatorPartitionWriter.scala

```diff
@@ -126,57 +129,63 @@ class CoordinatorPartitionWriter[T](
         val magic = logConfig.recordVersion.value
         val maxBatchSize = logConfig.maxMessageSize
         val currentTimeMs = time.milliseconds()
-
-        val recordsBuilder = MemoryRecords.builder(
-          ByteBuffer.allocate(math.min(16384, maxBatchSize)),
-          magic,
-          compressionType,
-          TimestampType.CREATE_TIME,
-          0L,
-          time.milliseconds(),
-          producerId,
-          producerEpoch,
-          0,
-          producerId != RecordBatch.NO_PRODUCER_ID,
-          RecordBatch.NO_PARTITION_LEADER_EPOCH
-        )
-
-        records.forEach { record =>
-          val keyBytes = serializer.serializeKey(record)
-          val valueBytes = serializer.serializeValue(record)
-
-          if (recordsBuilder.hasRoomFor(currentTimeMs, keyBytes, valueBytes, EMPTY_HEADERS)) recordsBuilder.append(
-            currentTimeMs,
-            keyBytes,
-            valueBytes,
-            EMPTY_HEADERS
-          ) else throw new RecordTooLargeException(s"Message batch size is ${recordsBuilder.estimatedSizeInBytes()} bytes " +
-            s"in append to partition $tp which exceeds the maximum configured size of $maxBatchSize.")
+        val bufferSupplier = threadLocalBufferSupplier.get()
+        val buffer = bufferSupplier.get(math.min(16384, maxBatchSize))
```

Comment:
Is this 16384 just the default batch size?

Reply:
That’s right. I actually took it from here: https://github.com/apache/kafka/blob/a83bc2d977d2af85d4edfc8096854137481001e9/core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala#L177